### DISCUSSION TITLE
Initialize each connection of a connection pool

### DISCUSSION THREAD

**Discussion by mkleen** [aio-libs/aiohttp#8993] [Q&A]
Hello,

I want to initialize each connection of my pool with a specific request to set server settings.  If I override `_create_connection` from `BaseConnector` I can control the connection creation:

```python
    async def _create_connection(self, req: ClientRequest, traces: List["Trace"], timeout: "ClientTimeout") -> ResponseHandler:
        
        transport, proto = await self._create_direct_connection(req, traces, timeout)
        
        return proto
```

How can I send a request to the newly created connection? Should i create a `ClientSession` inside `_create_connection` and post the request then? Is there a better way of doing this?

**Comment by Dreamsorcerer** [comment]
I'm not clear what you're trying to achieve, but I think you should be passing your custom connector to the ClientSession.

**Reply by mkleen** [reply]
I want to send a post request exactly once for each connection created. Once i created a `ClientSession` the connection pool is hidden and i can not distinguish the connections anymore.

**Reply by Dreamsorcerer** [reply]
You want to send a request in response to a connection being created, or you want a connection to be used for a single request only and then closed?

**Reply by mkleen** [reply]
I want to send a request in response to a connection being created.

**Reply by Dreamsorcerer** [reply]
Sounds like you want tracing. Glancing at the documentation, you want to use on_connection_create_end:
https://docs.aiohttp.org/en/stable/tracing_reference.html#aiohttp.TraceRequestStartParams

**Reply by mkleen** [reply]
Thanks a lot for the help. I will look into this.
